cmd/thanos/rule.go

```diff
@@ -564,6 +564,27 @@ func runRule(
 	return nil
 }
 
+// Scalar response consists of array with mixed types so it needs to be
+// unmarshaled separatelly.
+func convertScalarJSONToVector(scalarJSONResult json.RawMessage) (model.Vector, error) {
+	var resultPointSlice []json.RawMessage
+	var resultTime model.Time
+	var resultValue model.SampleValue
+	if err := json.Unmarshal(scalarJSONResult, &resultPointSlice); err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(resultPointSlice[0], &resultTime); err != nil {
```

Comment:
do you think could add additional check for `resultPointSlice` to be len of 2? That would avoid panics if the API return something else (:

Reply:
I'll definitely do that. Thanks.